test/distributed/test_device_mesh.py

```diff
@@ -50,6 +50,30 @@ def _set_env_var(addr="localhost", port="25364", world_size=1, rank=0):
     os.environ["RANK"] = f"{rank}"
 
 
+class WithCommsTest(DTensorTestBase):
+    """
+    Test with_comms to make sure the decorator works as expected with or without parenthesis.
+
+    #TODO: this is probably not the best place to put this test, but just keeping it so we have
+    test coverage for the with_comms.
+    """
+
+    @with_comms
+    def test_with_comms(self):
+        with self.assertRaises(AssertionError):
+            self.assertTrue(False)
+
+    @with_comms(eager_init=True)
+    def test_with_comms_eager_init(self):
+        with self.assertRaises(AssertionError):
+            self.assertTrue(False)
+
+    @with_comms()
+    def test_with_comms_non_eager_init(self):
+        with self.assertRaises(AssertionError):
+            self.assertTrue(False)
```

Comment:
I don't think we have any way to capture the "tests-not-being-run" error in CI, because if the tests are not being run, those tests will pass; and if tests are being run, those tests will still pass -- there's no differentiation.

Reply:
Besides, there's a legit error in tests. Please fix it before landing.